django/thunderstore/repository/migrations/0031_add_namespaces_to_existing_packages.py

```diff
@@ -0,0 +1,31 @@
+# Generated by Django 3.1.7 on 2021-11-04 20:32
+
+from django.db import migrations
+
+
+def forwards(apps, schema_editor):
+    Package = apps.get_model("repository", "Package")
+    packages = Package.objects.all()
+    for p in packages:
+        # Yes we assume here, there is only one Namespace currently linked to the Team
+        p.namespace = p.owner.namespaces.first()
+        p.save()
+
+
+def backwards(apps, schema_editor):
```

Comment:
I'm unsure if this is really required at all, but even if it was, you should definitely not go calling `.save()` on packages one by one. As previously you should use bulk updates and batching logic, also avoid fetching unnecessary data.

- Query IDs with `.values_list("id", flat=True).iterator()` to avoid fetching excessive amounts of data
- Batch the IDs into sets of 2000 each
- Use `.filter(id__in=ids).update(namespace=None)` for actually performing the update

Granted, we're not calling the `.save()` method so post-update hooks aren't triggered (e.g. cache busting), but that's fine and actually wanted in this case